Patch-Source: https://github.com/abseil/abseil-cpp/commit/6089180640a78417aae3032ffbbd73b4c3a6d053
--
From 6089180640a78417aae3032ffbbd73b4c3a6d053 Mon Sep 17 00:00:00 2001
From: Derek Mauro <dmauro@google.com>
Date: Thu, 27 Aug 2026 11:43:26 -0700
Subject: [PATCH] Fix SwisstableCollisions.LowEntropyStrings on architectures
 where Group::kWidth is less than 16 by scaling the number of expected probes
 inversely with Group::kWidth

Fixes #2142

PiperOrigin-RevId: 972071298
Change-Id: I222c42d57e89552301adcfd4fdfd2ed80f056138
---
 absl/hash/hash_test.cc | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/absl/hash/hash_test.cc b/absl/hash/hash_test.cc
index d20b2963..6048ce78 100644
--- a/absl/hash/hash_test.cc
+++ b/absl/hash/hash_test.cc
@@ -1305,6 +1305,10 @@ TEST(SwisstableCollisions, DoubleRange) {
 TEST(SwisstableCollisions, LowEntropyStrings) {
   constexpr char kMinChar = 0;
   constexpr char kMaxChar = 64;
+  // Scale the probe limit inversely with Group::kWidth so the test asserts a
+  // consistent bound on the number of probed slots across architectures.
+  constexpr size_t kMaxProbes =
+      64 * 16 / absl::container_internal::Group::kWidth;
   // These sizes cover the different hashing cases.
   for (size_t size : {8u, 16u, 32u, 64u, 128u}) {
     for (size_t b = 0; b < size - 1; ++b) {
@@ -1316,7 +1320,7 @@ TEST(SwisstableCollisions, LowEntropyStrings) {
           s[b + 1] = c2;
           set.insert(s);
           ASSERT_LT(HashtableDebugAccess<decltype(set)>::GetNumProbes(set, s),
-                    64)
+                    kMaxProbes)
               << "size: " << size << "; bit: " << b;
         }
       }
